core/templates/pages/exploration-player-page/modals/display-hint-modal.component.ts

```diff
@@ -33,12 +34,12 @@ import { PlayerTranscriptService } from '../services/player-transcript.service';
   templateUrl: './display-hint-modal.component.html'
 })
 export class DisplayHintModalComponent {
-  index: number;
-  COMPONENT_NAME_HINT: string;
-  hint: SubtitledHtml;
-  displayedCard: StateCard;
-  recordedVoiceovers: RecordedVoiceovers;
-  hintContentId: string;
+  index: number | undefined;
+  COMPONENT_NAME_HINT: string = AppConstants.COMPONENT_NAME_HINT;
+  hint: SubtitledHtml | null | undefined;
+  displayedCard: StateCard | undefined;
+  recordedVoiceovers: RecordedVoiceovers | undefined;
+  hintContentId: string | null | undefined;
```

Comment:
Can we limit the nulls and undefineds here? Either by using `!` or with other approach?

Reply:
! is a non-null and non-undefined assertion operator. Actually, index has been initialised in class DisplayHintModalComponent, and in the ngOnInit() function, value of index has not been set. So, index remains undefined even when `ngOnInit()` function is called. So, should we use `!` here as index is undefined here?

Reply:
@vojtechjelinek PTAL. Thanks!

Reply:
Not sure, maybe not for all of these, but can it be used for some of these?